app/controllers/api/v1/comments_controller.rb

```diff
@@ -0,0 +1,30 @@
+class Api::V1::CommentsController < ActionController::API
+  before_action :find_user_and_post
+
+  def index
+    @comments = @post.comments
+    render json: @comments
+  end
+
+  def create
+    @comment = @post.comments.new(comment_params)
+    @comment.author = @user
+
+    if @comment.save
+      render json: @comment, status: :created
+    else
+      render json: { error: @comment.errors.full_messages }, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:text)
+  end
+
+  def find_user_and_post
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:post_id])
+  end
```

Comment:
- [ ] Great work on the API so far, team! I wanted to provide some feedback regarding one of the requirements. According to the requirement, "the owner of the comment is the user that makes it; **it should not be set by passing a parameter**" However, I noticed that the `find_user_and_post` method retrieves the `@user` based on the `user_id` parameter passed in the URL params.
- To fulfill this requirement, you can leverage the `current_user` helper method defined in Devise. This method allows you to access the currently authenticated user. Keep up the good work, and let me know if you have any questions or need further assistance!

Reply:
Hi @AbelG101 Please i Will be glad if you can guide me through, Thank you

Reply:
Hello @katfogy sorry for the late reply but I'm just seeing this have you tried using the `current_user` from Devise?